python/tskit/trees.py

```diff
@@ -3070,6 +3070,117 @@ def simplify(
         else:
             return new_ts
 
+    def slice(
+            self, start=None, stop=None, reset_coordinates=True, simplify=True,
+            record_provenance=True):
+        """
+        Truncate this tree sequence, returning a tree sequence that spans a reduced
+        genomic interval. As with all tree sequence coordinates, the interval is treated
+        as half closed, that is, the leftmost point is included and rightmost point
+        excluded. Sites not covered by the sliced region are thrown away.
```

Comment:
This could be a bit more descriptive: it doesn't just return "a" tree sequence; it's a particular one. How about
```
        Returns the tree sequence containing only information in this tree sequence about the
        genomic interval between ``start`` and ``stop``. As with all tree sequence coordinates,
        the interval is treated
        as half closed, that is, the leftmost point is included and rightmost point
        excluded. Sites not covered by the sliced region are thrown away.
```

Reply:
Actually, I find it a bit confusing to refer to two different TSs, and it's probably unnecessary. We could simply say:

        Truncate this tree sequence to include only information in the genomic interval
        between ``start`` and ``stop``.  Edges are truncated to this interval, and sites
        not covered by the sliced region are thrown away.

It's obvious from the return values that we return a TS rather than truncating in-place (which is not possible anyway, given that tree sequences are immutable)